### ISSUE TITLE
Why you add a normalize layer on the head of resnet instead of using the normalize augment

### ISSUE THREAD

**Issue by junleiz** [alinlab/CSI#33]
Hi, Thank you for your excellent work. I noticed that you add a normalize layer on the head of resnet:
`class NormalizeLayer(nn.Module):
    """
    In order to certify radii in original coordinates rather than standardized coordinates, we
    add the Gaussian noise _before_ standardizing, which is why we have standardization be the first
    layer of the classifier rather than as a part of preprocessing as is typical.
    """

    def __init__(self):
        super(NormalizeLayer, self).__init__()

    def forward(self, inputs):
        return (inputs - 0.5) / 0.5`

`    def penultimate(self, x, all_features=False):
        out_list = []

        out = self.normalize(x)
        out = self.conv1(out)
        out = self.bn1(out)
        out = F.relu(out)
        out_list.append(out)

        out = self.layer1(out)
        out_list.append(out)
        out = self.layer2(out)
        out_list.append(out)
        out = self.layer3(out)
        out_list.append(out)
        out = self.layer4(out)
        out_list.append(out)

        out = F.avg_pool2d(out, 4)
        out = out.view(out.size(0), -1)

        if all_features:
            return out, out_list
        else:
            return out`

And normally we add a normalize augment in the dataloader like: 
`    normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],
                                     std=[0.229, 0.224, 0.225])

    train_dataset = datasets.ImageFolder(
        traindir,
        transforms.Compose([
            transforms.RandomResizedCrop(224),
            transforms.RandomHorizontalFlip(),
            transforms.ToTensor(),
            normalize,
        ]))
`
But I noticed that you did not use the normalize in the data augment . Furthermore, they are totally different(normalize vs normalize layer in your code)

**Comment by jihoontack** [comment]
Hi, thank you for the interest!

We used `NormalizeLayer` rather than `torchvision.transform.Normalize`, for efficient training.

We simply fixed the normalization mean and variance to 0.5 for all datasets (we believe, the result will be almost the same, as the batch normalization considers the data statistics).

Thank you
